generator/transformers/common_producer.py

```diff
@@ -0,0 +1,120 @@
+"""
+Common transformation
+"""
+
+import logging
+import re
+from abc import ABC
+from collections import namedtuple
+
+from model.array import Array
+from model.enum import Enum
+from model.struct import Struct
+
+
+class InterfaceProducerCommon(ABC):
+    """
+    Common transformation
+    """
+
+    version = '1.0.0'
+
+    def __init__(self, container_name, enums_package, structs_package, package_name,
+                 enum_names=(), struct_names=(), key_words=()):
+        self.logger = logging.getLogger('Generator.InterfaceProducerCommon')
+        self.container_name = container_name
+        self.enum_names = enum_names
+        self.struct_names = struct_names
+        self.key_words = key_words
+        self.enums_package = enums_package
+        self.structs_package = structs_package
+        self.package_name = package_name
+        self._params = namedtuple('params', 'deprecated description key last mandatory origin return_type since title '
+                                            'param_doc name')
+
+    @property
+    def get_version(self):
+        return self.version
+
+    @property
+    def params(self):
+        """
+        :return: namedtuple params(name='', origin='')
+        """
+        return self._params
+
+    @staticmethod
+    def key(param: str):
+        """
+        Convert param string to uppercase and inserting underscores
+        :param param: camel case string
+        :return: string in uppercase with underscores
+        """
+        if re.match(r'^[A-Z_\d]+$', param):
+            return param
+        else:
+            return re.sub(r'([a-z]|[A-Z]{2,})([A-Z]|\d$)', r'\1_\2', param).upper()
+
+    @staticmethod
+    def ending_cutter(n: str):
+        """
+        If string not contains only uppercase letters and end with 'ID' deleting 'ID' from end of string
+        :param n: string to evaluate and deleting 'ID' from end of string
+        :return: if match cut string else original string
+        """
+        if re.match(r'^\w+[a-z]+([A-Z]{2,})?ID$', n):
+            return n[:-2]
+        else:
+            return n
+
+    @staticmethod
+    def extract_description(d):
+        """
+        Evaluate, align and delete @TODO
+        :param d: list with description
+        :return: evaluated string
+        """
+        return re.sub(r'(\s{2,}|\n|\[@TODO.+)', ' ', ''.join(d)).strip() if d else ''
+
+    @staticmethod
+    def replace_sync(name):
+        """
+        :param name: string with item name
+        :return: string with replaced 'sync' to 'Sdl'
+        """
+        if name:
+            return re.sub(r'^([sS])ync(.+)$', r'\1dl\2', name)
+        return name
+
+    def replace_keywords(self, name: str = '') -> str:
```

Comment:
The comparison in this method seems to be case sensitive. It should be insensitive.

Reply:
@bilal-alsharifi restored case insensitive matching, and also fixed functionID parameter replacement, thanks